packages/universal-sql/src/build-parquet.js

```diff
@@ -0,0 +1,54 @@
+import {
+	tableFromArrays,
+	tableToIPC,
+	vectorFromArray,
+	Float64,
+	Utf8,
+	Timestamp,
+	Bool
+} from 'apache-arrow';
+import { Compression, writeParquet, WriterPropertiesBuilder } from 'parquet-wasm/node/arrow1.js';
+
+/**
+ * Constructs a Buffer from provided table metadata
+ * @param {{name: string, evidenceType: string}[]} columns
+ * @param {any[]} data
+ * @returns {Promise<Uint8Array>}
+ */
+export async function buildParquetFromResultSet(columns, data) {
+	const m = columns.map((c) => {
+		const rawValues = data.map((d) => d[c.name] ?? `Null`);
+
+		// TODO: Make this work properly.
+
+		/** @type {import("apache-arrow").Vector} */
+		let values;
+		if (c.evidenceType === 'number') {
+			values = vectorFromArray(rawValues, new Float64());
+		} else if (c.evidenceType === 'string') {
+			values = vectorFromArray(rawValues, new Utf8());
+		} else if (c.evidenceType === 'date') {
+			values = vectorFromArray(rawValues, new Timestamp());
```

Comment:
This should be `new DateMillisecond()`

Reply:
Also might break if passed a stringified date type instead of a `Date` object, which I think some adapters do - might be worth shifting it to be the responsibility of the database adapter to return a `Date` object?

Reply:
> might be worth shifting it to be the responsibility of the database adapter to return a Date object?

This is probably a good way to go; do you have docs for DateMillisecond? Apache Arrow docs had examples set up this way

Reply:
No docs, but only the `DateDay` and `DateMillisecond` class extend the `Date_` class from what I can tell, and the type of `vectorFromArray` that accepts `Date` objects requires that